### ISSUE TITLE
[v2] Volume mounts are empty / sshfs fails to connect when node specs do not include podCIDR(s)

### ISSUE THREAD

**Issue by jkernagis2** [telepresenceio/telepresence#1779]
Original Slack thread: https://datawire-oss.slack.com/archives/CAUBBJSQZ/p1623100957271700

tl;dr - telepresence 2 intercept seems to work, but not volume mounts.

The following logs seem to be of note in the connector.log file:

```
2021/06/07 15:13:51.1303 debug   connector/background-manager/intercept-port-forward/***:41969 : sshfs waiting 6s before retrying after error: dial tcp ***:41969: i/o timeout (from pkg/client/retry.go:48)
2021/06/07 15:14:00.1341 debug   connector/background-manager/intercept-port-forward/***:41969 : sshfs waiting 6s before retrying after error: dial tcp ***:41969: i/o timeout (from pkg/client/retry.go:48)
2021/06/07 15:14:09.1382 debug   connector/background-manager/intercept-port-forward/***:41969 : sshfs waiting 6s before retrying after error: dial tcp ***:41969: i/o timeout (from pkg/client/retry.go:48)
```

I am able to connect via telnet from traffic manager to that IP/port - the following is from a shell attached to the traffic-manager pod, using the same IP/port that's being logged in connector.log above:

```
/ # echo $HOSTNAME
traffic-manager-c9dcccc4-hkpqg
/ # apk add busybox-extras
(1/1) Installing busybox-extras (1.32.1-r6)
Executing busybox-extras-1.32.1-r6.post-install
Executing busybox-1.32.1-r6.trigger
OK: 7 MiB in 17 packages
/ # busybox-extras telnet *** 41969
Connected to ***
```

While debugging with Thomas Hallgren on the telepresence slack, we've found that the following commands have no return value:

```
k get nodes -o jsonpath={..spec.podCIDR}
k get nodes -o jsonpath={..spec.podCIDRs}
```

Context from Slack / Thomas:

```
Thomas Hallgren  41 minutes ago
Sorry, what I’m really after is this: k get nodes -o jsonpath={..spec.podCIDR}

John Kernagis  41 minutes ago
I get no output for that one :thinking_face:

Thomas Hallgren  40 minutes ago
now that’s interesting.

Thomas Hallgren  39 minutes ago
Those CIDRs are what telepresence make available in the TUN-device. Without them, there won’t be much that works. So we need to figure out how to retrieve the correct subnets some other way.

Thomas Hallgren  38 minutes ago
What about this: k get nodes -o jsonpath={..spec.podCIDRs} (note the extra ‘s’ at the end)

John Kernagis  36 minutes ago
also no output unfortunately

Thomas Hallgren  27 minutes ago
I guess that must mean that a global cluster CIDR is used. (edited)
```

```
Thomas Hallgren  10 minutes ago
We need to understand how to obtain the podCIDR(s) for systems that don’t include that in the nodes.
```

**Comment by thallgren** [comment]
We will have at least a workaround for this when #1776 is released because it will give the user a chance to manually enter the relevant podCIDR's in the config.

**Comment by efunk** [comment]
Delivered in the Telepresence 2.3.1 GA release.

**Comment by jkernagis2** [comment]
@efunk Assuming what was Delivered is the "also-proxy" option from #1776 , I'm not sure this should have been closed given that was as far as I'm aware just a workaround for this issue.  Importantly, see @thallgren 's point: "We need to understand how to obtain the podCIDR(s) for systems that don’t include that in the nodes."

If this workaround ends up being THE way to get things to work when nodes don't include the podCIDR(s), shouldn't there be an update to the telepresence 2 documentation mentioning this situation, and giving the "also-proxy" option from #1776 as the fix?

**Comment by thallgren** [comment]
Reopening since #1776 is merely a way to work around the problem. We will come up with a proper solution where telepresence is able to find those subnets.

**Comment by efunk** [comment]
Resolved in Telepresence 2.3.2 GA

**Comment by mtrimolet** [comment]
_(Copied from slack, 21/09/01)_
Hello ! I’ve been having a similar issue, i.e. sshfs throws an “i/o timeout” in _connector.log_, and the mounted volume appears empty. But with some varying detail though :
- the `podCIDR(s)` commands are properly returning a subnet per node, but the ip used by sshfs matches none of these subnets (though it answers to telnet, as in original issue)

- interesting logs regarding DNS :
  - _traffic-manager_’s log contains a line with this error : 
  ```
  [...] LookupHost response zpbtzxyp.<some-namespace>.svc.cluster.local -> NOT FOUND [...]
  [...] LookupHost response <my-local-hostname>.<some-namespace>.svc.cluster.local -> NOT FOUND [...]
  ```
  - a similar line at beginning of daemon.log containing 
  ```
  [...] QTYPE[1] zpbtzxyp.cluster.local. -> NOT FOUND [...]
  [...] QTYPE[1] <my-local-hostname>.cluster.local. -> NOT FOUND [...]
  ```
  - and a random combination of all these different variants of addresses in _daemon.log_ once I try to `intercept`
- _traffic-manager_’s log contains another interesting line at beginning : 
```
level=error msg="unable to get `default` namespace: namespaces \"default\" is forbidden: User \"system:serviceaccount:ambassador:traffic-manager\" cannot get resource \"namespaces\" in API group \"\" in the namespace \"default\", using default clusterID: 00000000-0000-0000-0000-000000000000"
```
I hope some of these infos can be of any use to the issue, anyway I’ve since downgraded to telepresence-legacy and it works well.
Note : `<some-namespace>` is the namespace I work with (intercept included), trafic-manager is installed in `ambassador` and `default` is empty. `zpbtzxyp` is an exemple but many similar ids are present, and they change when I uninstall everything and install again

Edit: solved with `podCIDRStrategy: coverPodIPs` (#2135), caused by configuration mismatch between configured `podCIDRs` and effective pod IPs
